compilers/concrete-compiler/compiler/lib/Runtime/simulation.cpp

```diff
@@ -183,6 +189,29 @@ void sim_wop_pbs_crt(
 
 uint64_t sim_neg_lwe_u64(uint64_t plaintext) { return ~plaintext + 1; }
 
+uint64_t sim_add_lwe_u64(uint64_t lhs, uint64_t rhs, char *loc,
+                         bool is_signed) {
+  uint64_t result = lhs + rhs;
+  if (!is_signed && (lhs > UINT63_MAX - rhs || result > UINT63_MAX)) {
```

Comment:
You should also check when is_signed is true. And I not understand why you are checking the result, the first test should be enough?
So as we discuss you need to check for underflow too.